main.js

```diff
@@ -172,6 +172,12 @@ app.on('window-all-closed', function () {
   }
 })
 
+app.on('activate', function() {
+  if (!Object.keys(windows).length) {
+    init()
```

Comment:
Actually it seems that hasVisibleWindows is not necessary if you are checking Object.keys(windows).length already. I'm not super familiar with this API.